includes/class-wc-calypso-bridge-coming-soon.php

```diff
@@ -80,19 +95,88 @@ public function should_exclude_lys_coming_soon( $exclude ) {
 	 *
 	 * @param int $old_value
 	 * @param int $new_value
-	 * @param string $option
 	 * @return void
 	 */
-	public function sync_coming_soon_option( $old_value, $new_value, $option ) {
+	public function sync_coming_soon_wpcom_to_lys( $old_value, $new_value ) {
 		if ( ! class_exists( '\Automattic\WooCommerce\Admin\Features\Features' ) || ! \Automattic\WooCommerce\Admin\Features\Features::is_enabled( 'launch-your-store' ) ) {
 			return;
 		}
 
+		$woocommerce_coming_soon = get_option( 'woocommerce_coming_soon', false );
+		$is_coming_soon_wpcom = 1 === (int) $new_value;
+
+		// Value is already set, we don't need to update option.
```

Comment:
If I'm not mistaken, I think this may not be necessary since `update_option` will check if the value is already set and, if it is, not update the option. But up to you to decide whether to keep it or not.

Reply:
Good catch! I gave it a thought, and I think it may still be useful to exit early since we're doing hook shenanigans below.